modules/tinymce/src/plugins/table/main/ts/ui/UiUtils.ts

```diff
@@ -51,10 +51,35 @@ const generateItemsCallback = <T extends Item>(editor: Editor, items: T[], forma
   (callback: (items: Menu.ToggleMenuItemSpec[]) => void) =>
     callback(generateItems(editor, items, format, extractText, onAction));
 
+const fixColorValue = (value: string, setColor: (colorValue: string) => void) => {
+  if (value === 'remove') {
+    setColor('');
+  } else {
+    setColor(value);
+  }
+};
+
+const generateColorSelector = (editor: Editor, colorList: Menu.ChoiceMenuItemSpec[], style: string): Menu.FancyMenuItemSpec[] => [
+  {
+    type: 'fancymenuitem',
+    fancytype: 'colorswatch',
+    initData: {
+      colors: colorList.length > 0 ? colorList : undefined,
+      disallowCustomColors: true
+    },
+    onAction: (data) => {
+      fixColorValue(data.value, (value: string) => {
+        editor.execCommand('mceTableApplyCellStyle', false, { [style]: value });
+      });
+    }
+  }
+];
```

Comment:
```suggestion
const generateColorSelector = (editor: Editor, colorList: Menu.ChoiceMenuItemSpec[], style: string): Menu.FancyMenuItemSpec[] => [{
  type: 'fancymenuitem',
  fancytype: 'colorswatch',
  initData: {
    colors: colorList.length > 0 ? colorList : undefined,
    disallowCustomColors: true
  },
  onAction: (data) => {
    fixColorValue(data.value, (value) => {
      editor.execCommand('mceTableApplyCellStyle', false, { [style]: value });
    });
  }
}];
```
We don't need the explicit `string` type here and we also can condense it a little more since this will only ever have one value in the array.